src/nxdoc/nuxeo-server/runtime-and-component-model/runtime-and-component-model-tutorials/how-to-create-a-service.md

```diff
@@ -0,0 +1,417 @@
+      <display>
+        <on>MyDisplay1</on>
+        <on>MyDisplay2</on>
+      </display>
+      <properties>
+        <property name="MyPropName1">MyPropValue1</property>
+        <property name="MyPropName2">MyPropValue2</property>
+      </properties>
+    </exporter>
+    </extension>
+</component>
+```
+
+## {{> anchor 'defining-properties'}}Defining Properties on the Service Component
+
+Defining an extension point is not strictly necessary if you just want to expose a service and do not need configurable elements. If some simple elements need to be configurable, there are several options:
+1.  Define a property on the service. Note that this property value **will not be able to be overridden** by another contribution, but it might be useful to keep that value in the XML component declaration:
+    ```
+    <?xml version="1.0"?>
+    <component name="org.mycompany.myproject.MyService">
+
+      <implementation class="org.mycompany.myproject.MyComponent" />
+      <property name="myProp" value="myValue" />
+      <!-- previous configuration here... -->
+
+    </component>
+    ```
+
+    From the component code, that property can be retrieved from the Runtime context, when activating the component:
+    ```
+    public class MyComponent extends DefaultComponent {
+
+        @Override
+        public void activate(ComponentContext context) {
+            String myPropValue = (String) context.getPropertyValue("myProp");
+        }
+
+    }
+    ```
+1.  Contribute a property to the default service `ConfigurationService`. This property can be defined alongside the component and service declarations. Note that this property value **will be able to be overridden** by another contribution, and will be able to be documented specifically:
+    ```
+    <?xml version="1.0"?>
+    <component name="org.mycompany.myproject.MyService">
+
+      <implementation class="org.mycompany.myproject.MyComponent" />
+      <!-- previous configuration here... -->
+
+      <extension target="org.nuxeo.runtime.ConfigurationService" point="configuration">
+        <documentation>
+          This configuration property can be documented here. HTML tags are accepted.
+        </documentation>
+        <property name="org.mycompany.myproject.myprop">myDefaultValue</property>
+      </extension>
+
+    </component>
+    ```
+
+    From the component code, that property can be retrieved from the Runtime framework, after starting the component, for instance (to make sure ConfigurationService is available):
+    ```
+    import org.nuxeo.runtime.services.config.ConfigurationService;
+
+    public class MyComponent extends DefaultComponent {
+
+        protected static final String MY_PROP_NAME = "org.mycompany.myproject.myprop";
+
+        @Override
+        public void start(ComponentContext context) {
+            super.start(context);
+            String myPropValue = Framework.getService(ConfigurationService.class).getString(MY_PROP_NAME);
```

Comment:
`getString` method returns an _Optional<String>_, we may want to use the signature with the default parameter?